qiime/plugin/testing.py

```diff
@@ -84,3 +90,33 @@ def assertSemanticTypeRegisteredToFormat(self, semantic_type, exp_format):
             obs_format, exp_format,
             "Expected semantic type %r to be registered to format %r, not %r."
             % (semantic_type, exp_format, obs_format))
+
+    def transform_format(self, source_format, target, **kwargs):
```

Comment:
Can you replace `**kwargs` with `filename=None, filenames=None` here? That way the function signature is explicit about what we are up to here.